src/popup/pages/Retip.vue

```diff
@@ -140,30 +135,43 @@ export default defineComponent({
           : AETERNITY_COIN_PRECISION,
       ).toNumber();
       loading.value = true;
-      await watchUntilTruthy(() => tippingV1.value);
       try {
+        const { tippingV1, tippingV2 } = await getTippingContracts();
+        const tippingContract = tipId.includes('_v2') || tipId.includes('_v3')
+          ? tippingV2
+          : tippingV1;
+        if (!tippingContract) {
+          throw Error('failed to initialize tipping contract');
+        }
         let retipResponse = null;
-        if (formModel.value.selectedAsset?.contractId !== AETERNITY_CONTRACT_ID) {
+        if (
+          tippingV2
+          && formModel.value.selectedAsset?.contractId
+          && formModel.value.selectedAsset.contractId !== AETERNITY_CONTRACT_ID
+        ) {
           await root.$store.dispatch(
             'fungibleTokens/createOrChangeAllowance',
             [
-              formModel.value.selectedAsset?.contractId,
+              formModel.value.selectedAsset.contractId,
               formModel.value.amount],
           );
 
-          retipResponse = await tippingV2.value.methods.retip_token(
+          retipResponse = await tippingV2.retip_token(
             +tip.value.id.split('_')[0],
-            formModel.value.selectedAsset?.contractId,
+            formModel.value.selectedAsset.contractId as any,
```

Comment:
Doesn't it have `string` type?

Reply:
But it awaits a `ct${string}` and applying such in the `contractId` field natively resulting in a big refactoring, that i'm planning to do separatly.